google-account-plugin/src/com/google/cloud/tools/intellij/stats/GoogleUsageTracker.java

```diff
@@ -36,64 +34,76 @@
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.impl.client.HttpClientBuilder;
 import org.apache.http.message.BasicNameValuePair;
-import org.jetbrains.annotations.NonNls;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Google Usage Tracker that reports to Cloud Tools Analytics backend.
  */
 public class GoogleUsageTracker implements UsageTracker {
 
-  private static final Logger LOG = Logger.getInstance(GoogleUsageTracker.class);
-  @NonNls
+  private static final Logger logger = Logger.getInstance(GoogleUsageTracker.class);
+
   private static final String ANALYTICS_URL = "https://ssl.google-analytics.com/collect";
-  @NonNls
-  private static final String ANALYTICS_APP = "Cloud Tools for IntelliJ";
-  @NonNls
-  private static final String INTELLIJ_EDITION = "intellij.edition";
+  private static final String ANALYTICS_APP = "gcloud-intellij";
 
   private final String analyticsId;
 
   public GoogleUsageTracker() {
     this.analyticsId = UsageTrackerManager.getInstance().getAnalyticsProperty();
   }
 
-  private static final List<? extends NameValuePair> analyticsBaseData = ImmutableList
-      .of(new BasicNameValuePair("v", "1"),
-          new BasicNameValuePair("t", "event"),
-          new BasicNameValuePair("an", ANALYTICS_APP),
-          new BasicNameValuePair("av", ApplicationInfo.getInstance().getFullVersion()),
-          new BasicNameValuePair("cid",
+  private static final List<BasicNameValuePair> analyticsBaseData = ImmutableList
+      .of(new BasicNameValuePair("v", "1"),         // Protocol version
+          new BasicNameValuePair("t", "pageview"),  // Note the "pageview" type, not "event"
+          new BasicNameValuePair("ni", "0"),        // Non-interactive? Report as interactive.
+          new BasicNameValuePair("cid",             // UUID for this IntelliJ client
               UpdateChecker.getInstallationUID(PropertiesComponent.getInstance())));
 
+  /**
+   * Send a (virtual) "pageview" ping to the Cloud-platform-wide Google Analytics Property.
+   *
+   * @param eventCategory Ignored for plugins installed on IntelliJ. Retained to maintain backward
+   *     compatibility on Android Studio.
+   * @param eventAction On IntelliJ, this is the primary event action.
+   * @param eventLabel On IntelliJ, reported as a metadata key.
+   * @param eventValue On IntelliJ, reported as a metadata value.
+   */
   @Override
   public void trackEvent(@NotNull String eventCategory,
       @NotNull String eventAction,
       @Nullable String eventLabel,
       @Nullable Integer eventValue) {
+    if (eventValue != null) {
```

Comment:
whats the point of this if statement if you are using preconditions#checkArgument right after?